token/typed.go

```diff
@@ -0,0 +1,59 @@
+package token
+
+import (
+	"fmt"
+	"sort"
+)
+
+// ArgumentsParser defines a parser for the arguments of a typed token.
+// Parsing stops unrecoverably at the first error. The return value of Err must be checked before using the values returned by precedings calls to GetInt.
+type ArgumentsParser interface {
+	// GetInt tries to parse the argument name and returns its integer value or defaultValue it the argument is not found.
+	// The return value is valid only if Err returns nil.
+	GetInt(name string, defaultValue int) int
+	// Err returns the first error encountered by the ArgumentsParser.
+	Err() error
+}
+
+// CreateTyped defines a function to create a typed token
+type CreateTyped func(argParser ArgumentsParser) (Token, error)
```

Comment:
Yes, both is true. I guess golint would scream at this. Maybe a new new package would be better but I have two concerns:
- I fear that this will be the only code in the new package. which is not necessarily bad, but
- Alle the other packages under /token hold as you can see real tokens. So doing this would break this pattern.